### PR TITLE
feat: add `NFTs` and `POAPs` in wallet profile and add NFT Detail page

### PR DESCRIPTION
**Pull request by LeifXu**
# Pull Request

## Related Issue

Closes FW-948 FW-949 FW-1055 FW-1056

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/providers/types/Firefly.ts [hunk 1]
```diff
@@ -397,3 +399,272 @@ export type BlockRelationResponse = Response<
...
+    image_url: string;
+    image_properties: ImageProperties;
+    video_url: any;
+    video_properties: any;
+    audio_url: any;
+    audio_properties: any;
+    model_url: any;
+    model_properties: any;
+    other_url: any;
+    other_properties: any;
```
Comment (guanbinrui) [inline src/providers/types/Firefly.ts:643]: Avoid using `any`.

File: src/components/Profile/NFTCollectionList.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    return (
+        <div
+            className="flex cursor-pointer flex-col rounded-lg bg-lightBg p-1.5 pb-1 sm:rounded-2xl"
+            onClick={() => onClick?.(collection)}
+        >
+            <div
+                className={classNames('relative grid aspect-square h-auto w-full gap-1.5 overflow-hidden', {
+                    'grid-cols-2': images.length > 1,
+                    'grid-rows-2': images.length > 1,
```
Comment (swkatmask) [inline src/components/Profile/NFTCollectionList.tsx:44]:
```suggestion
                    'grid-cols-2 grid-rows-2': images.length > 1,
```

File: src/components/Profile/NFTCollectionList.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+export function NFTCollectionItem({ collection, onClick }: NFTCollectionItemProps) {
+    const distinctNFTCount = collection.collection_details.distinct_nft_count;
+    const images = useMemo(() => {
+        if (!collection.nftPreviews) return [];
+        let previewImages = collection.nftPreviews;
+        if (collection.nftPreviews.length === 4 && distinctNFTCount === 4) {
+            previewImages = previewImages?.slice(0, 4);
+        } else {
+            previewImages = previewImages?.slice(0, 3);
```
Comment (swkatmask) [inline src/components/Profile/NFTCollectionList.tsx:27]:
```suggestion
            previewImages = previewImages.slice(0, 4);
        } else {
            previewImages = previewImages.slice(0, 3);
```
It has already been checked above.

File: src/app/(normal)/nft/[address]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+const tabs = [
+    {
+        label: t`Items`,
+        value: 'items',
+    },
+    {
+        label: t`Top Collectors`,
+        value: 'topCollectors',
+    },
+] as const;
```
[the remark this answers is not in the document]
Comment (LeifXu) [reply src/app/(normal)/nft/[address]/page.tsx:29]: It is a constant, so there is no need to update it

File: src/providers/types/Firefly.ts [hunk 5]
```diff
@@ -428,3 +430,264 @@ export type BlockRelationResponse = Response<
...
+    value_usd_cents: number;
+}
+
+export interface CollectionRoyalty {
+    source: string;
+    total_creator_fee_basis_points: number;
+    recipients: Recipient[];
+}
+
+export interface Collection2 {
```
Comment (guanbinrui) [inline src/providers/types/Firefly.ts:601]: Use a better name.

File: src/providers/types/Firefly.ts [hunk 6]
```diff
@@ -428,3 +430,264 @@ export type BlockRelationResponse = Response<
...
+    external_url?: string;
+    twitter_username?: string;
+    discord_url?: string;
+    instagram_username: any;
+    medium_username?: string;
+    telegram_url?: string;
+    marketplace_pages: MarketplacePage[];
+    metaplex_mint: any;
+    metaplex_candy_machine: any;
+    metaplex_first_verified_creator: any;
```
Comment (guanbinrui) [inline src/providers/types/Firefly.ts:483]: Avoid using name.

File: src/providers/types/Firefly.ts [hunk 7]
```diff
@@ -428,3 +430,264 @@ export type BlockRelationResponse = Response<
...
```
Comment (guanbinrui) [inline src/providers/types/Firefly.ts:619]: Avoid using any.

File: src/components/NFTDetail/NFTOverflow.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+export function EVMExplorerLink(props: { address: string; chainId?: number; type: 'address' | 'tx' }) {
+    if (props.chainId) {
+        const resolveExplorerLink = {
+            address: EVMExplorerResolver.addressLink.bind(EVMExplorerResolver),
+            tx: EVMExplorerResolver.transactionLink.bind(EVMExplorerResolver),
+        }[props.type];
+        return (
+            <span className="break-all">
+                <a
```
Comment (guanbinrui) [inline src/components/NFTDetail/NFTOverflow.tsx:40]: Link

File: src/app/(normal)/nft/[address]/page.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (guanbinrui) [reply src/app/(normal)/nft/[address]/page.tsx:29]: It should move inside the component.

File: src/app/(normal)/nft/[address]/[tokenId]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        return `${formatAmount(firstFloorPrice.value, -firstFloorPrice.payment_token.decimals)} ${
+            firstFloorPrice.payment_token.symbol
+        }`;
+    }, [data]);
+
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    if (error || !data) {
```
Comment (guanbinrui) [inline src/app/(normal)/nft/[address]/[tokenId]/page.tsx:50]:
```suggestion
    if (error || !data?.metadata) {
```

File: src/app/(normal)/nft/[address]/page.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        return `${formatAmount(firstFloorPrice.value, -firstFloorPrice.payment_token.decimals)} ${
+            firstFloorPrice.payment_token.symbol
+        }`;
+    }, [data?.floor_prices]);
+
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    if (error) {
```
Comment (guanbinrui) [inline src/app/(normal)/nft/[address]/page.tsx:54]:
```suggestion
    if (error || !data) {
```

File: src/app/(normal)/nft/[address]/page.tsx [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+                            {tab.label}
+                        </Tab>
+                    ))}
+                </Tabs>
+                {
+                    {
+                        items: <NFTList address={address} />,
+                        topCollectors: <TopCollectors address={address} totalQuantity={data?.total_quantity} />,
+                    }[currentTab]
+                }
```
Comment (guanbinrui) [inline src/app/(normal)/nft/[address]/page.tsx:88]: Use if-else stead.

File: src/app/(normal)/nft/[address]/page.tsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+        },
+    });
+
+    const floorPrice = useMemo(() => {
+        const firstFloorPrice = first(data?.floor_prices);
+        if (!firstFloorPrice) return;
+        return `${formatAmount(firstFloorPrice.value, -firstFloorPrice.payment_token.decimals)} ${
+            firstFloorPrice.payment_token.symbol
+        }`;
+    }, [data?.floor_prices]);
```
Comment (guanbinrui) [inline src/app/(normal)/nft/[address]/page.tsx:48]: A repeat part.

File: src/components/NFTDetail/NFTInfo.tsx [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+export function NFTInfo(props: NFTInfoProps) {
+    const { imageURL, name, tokenId, collection, ownerAddress, contractAddress, floorPrice, lastSale } = props;
+    return (
+        <div className="flex flex-col gap-5 sm:flex-row">
+            <div className="w-full sm:w-[300px]">
+                <Image width={300} height={300} src={imageURL} alt={name} className="rounded-[20px] object-cover" />
+                <div className="mt-4 flex justify-center gap-2">
+                    <a
```
Comment (guanbinrui) [inline src/components/NFTDetail/NFTInfo.tsx:32]: Link

File: src/components/Profile/NFTCollectionList.tsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+                        key={`${url}-${i}`}
+                        width={500}
+                        height={500}
+                        className="h-full w-full rounded-lg object-cover"
+                        src={url}
+                        alt="nft_image"
+                    />
+                ))}
+                {images.length === 3 && distinctNFTCount !== 4 && distinctNFTCount > 3 ? (
+                    <div className="flex h-full w-full items-center justify-center rounded-lg bg-white text-xs font-normal leading-4 dark:bg-[#202020] sm:rounded-xl">
```
Comment (guanbinrui) [inline src/components/Profile/NFTCollectionList.tsx:57]: Use a color from palette.

File: src/components/Tabs/index.tsx [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export function Tab({ children, value }: { children: ReactNode; value: string }) {
+    const { value: currentTab, onChange } = useContext(TabContext);
+
+    return (
+        <li className="flex flex-1 list-none justify-center lg:flex-initial lg:justify-start">
+            <a
+                className={classNames(
+                    currentTab === value ? 'border-b-2 border-[#9250FF] text-main' : 'text-third',
```
Comment (guanbinrui) [inline src/components/Tabs/index.tsx:49]: Use a color from palette.

File: src/components/NFTDetail/NFTOverflow.tsx [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+                    href={resolveExplorerLink(props.chainId, props.address)}
+                    target="_blank"
+                    className="text-[#9250FE] hover:underline"
```
Comment (guanbinrui) [inline src/components/NFTDetail/NFTOverflow.tsx:43]: Use a color from palette.

File: src/app/(normal)/nft/[address]/page.tsx [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+                {
+                    label: t`Items`,
+                    value: 'items',
+                },
+                {
+                    label: t`Top Collectors`,
+                    value: 'topCollectors',
+                },
+            ] as const,
+        [],
```
Comment (swkatmask) [inline src/app/(normal)/nft/[address]/page.tsx:32]:
```suggestion
        [t`Items`, t`Top Collectors`],
```

**Review by swkatmask — APPROVED** [review]

**Pull request merged**